### PR TITLE
Update Prettier

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Cleans up some changes around Prettier v3
- Notably, moves our `.prettierrc.cjs` file to a normal `prettier.config.js` file (requires `"type": "module"` in the root as well. **Is this a problem?**
- Updates our `format:code` script to actually format `.astro` files. For some reason, `prettier -w .` does not pick up on `.astro` files, but `prettier -w "**/*"` does.

## Testing

N/A, infra change only

## Docs

N/A, infra change only

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +15,7 @@
     "dev": "turbo run dev --concurrency=40 --parallel --filter=astro --filter=create-astro --filter=\"@astrojs/*\" --filter=\"@benchmark/*\"",
     "format": "pnpm run format:code",
     "format:ci": "pnpm run format:imports && pnpm run format:code",
-    "format:code": "prettier -w . --cache",
+    "format:code": "prettier -w \"**/*\" --ignore-unknown --cache",
```
[the remark this answers is not in the document]
Comment (natemoo-re) [reply package.json:18]: Turns out this an upstream issue! https://github.com/prettier/prettier/issues/15079

**Comment by natemoo-re** [comment]
See also https://github.com/withastro/astro/pull/8280

File: .prettierignore [hunk 2]
```diff
@@ -1,10 +1,3 @@
-# Ignoring these files for now due to https://github.com/withastro/compiler/issues/585
-examples/basics/src/layouts
-examples/with-nanostores/src/layouts
-
-# Ignoring this file for now due to https://github.com/withastro/compiler/issues/485
-packages/markdown/component/Markdown.astro
-
```
[the remark this answers is not in the document]
Comment (Princesseuh) [reply .prettierignore:7]: One of my first contributions to Astro 🫡

File: package.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "root",
+  "type": "module",
```
[the remark this answers is not in the document]
Comment (bluwy) [reply package.json:3]: All of our subfolders have this config too, so I think it should be safe. The only other JS file in root is `eslintrc.cjs`, but that's already explicitly cjs.

**Review by bluwy — APPROVED** [review]

**Review by Princesseuh — APPROVED** [review]

**Pull request merged**